### PR TITLE
Cody: Publish embedding error message

### PR DESCRIPTION
**Pull request by deepak2431**
This PR is for issue #52480. It implements the method to publish two different types of embeddings error:
1. Connection Error
2. Search results error

## Test plan
All unit tests have passed.
Tested it by making the condition TRUE, and passing a dummy error message.

### CODE REVIEW

File: client/cody/src/chat/ChatViewProvider.ts [hunk 1]
```diff
@@ -650,11 +651,27 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
...
-
         this.disposables.push(vscode.window.onDidChangeTextEditorSelection(() => send()))
         send()
     }
 
+    /**
+     * Publish embedding connections or results error to webview
+     */
+    private publishEmbeddingsError(): void {
+        if (!this.codebaseContext.checkEmbeddingsConnection()) {
```
Comment (abeatrix) [inline client/cody/src/chat/ChatViewProvider.ts:662]: I think this is great! I'd just combine this with the sendErrorToWebview function and call it in onCompletionEnd() that way we can always surface the latest errors from all services, what do you think?

**Review by abeatrix — CHANGES_REQUESTED** [review]
Left comments inline but I think this is a great approach! Thank you so much for the quick solution here ❤️

File: client/cody/src/chat/ChatViewProvider.ts [hunk 2]
```diff
@@ -410,6 +410,7 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
         }
 
         this.codebaseContext = codebaseContext
+        this.publishEmbeddingsError()
```
[the remark this answers is not in the document]
Comment (deepak2431) [reply client/cody/src/chat/ChatViewProvider.ts:413]: Agreed with you here. I was also not 100% sure what would be the best place to call this. So on bot response would be good so any error in between can also be displayed.

File: client/cody/src/chat/ChatViewProvider.ts (hunk 1 above)
Comment (deepak2431) [reply client/cody/src/chat/ChatViewProvider.ts:662]: Sorry, I didn't get what you meant here.  Combining with sendErrorToWebview function and calling it in onCompletionEnd() ??

Comment (deepak2431) [reply client/cody/src/chat/ChatViewProvider.ts:662]: Okay, I see what you mean. To check, the embedding connection error should be called onCompletionEnd(), and the search errors should be displayed with onBotResponse.

File: client/cody/src/chat/ChatViewProvider.ts [hunk 3]
```diff
@@ -362,6 +363,12 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
         this.sendTranscript()
         void this.saveTranscriptToChatHistory()
         void vscode.commands.executeCommand('setContext', 'cody.reply.pending', false)
+        if (!this.codebaseContext.checkEmbeddingsConnection()) {
```
Comment (abeatrix) [inline client/cody/src/chat/ChatViewProvider.ts:366]:
```suggestion
        this.publishEmbeddingsError()
```

File: client/cody/src/chat/ChatViewProvider.ts [hunk 4]
```diff
@@ -650,11 +657,21 @@ export class ChatViewProvider implements vscode.WebviewViewProvider, vscode.Disp
...
-
         this.disposables.push(vscode.window.onDidChangeTextEditorSelection(() => send()))
         send()
     }
 
+    /**
+     * Publish embedding connections or results error to webview
+     */
+    private publishEmbeddingsError(): void {
+        const searchErrors = this.codebaseContext.getEmbeddingSearchErrors()
```
Comment (abeatrix) [inline client/cody/src/chat/ChatViewProvider.ts:668]:
```suggestion
        const searchErrors = this.codebaseContext.getEmbeddingSearchErrors()
        if (searchErrors.length) {
                this.sendErrorToWebview(searchErrors)
        }
```

**Review by abeatrix — APPROVED** [review]
Added some new suggestions but what you have is fine too! If you're happy with it just lmk and I can start with the merging process!

**Comment by deepak2431** [comment]
@abeatrix Let's make this best, so made all the suggested changes. All good to merge now!

**Comment by abeatrix** [comment]
Perfect! Thank you so much! I will start merging this by end of today, if not, I will do this first thing in the morning on Tuesday. Will keep you posted!

**Comment by deepak2431** [comment]
> Perfect! Thank you so much! I will start merging this by end of today, if not, I will do this first thing in the morning on Tuesday. Will keep you posted!

Sounds great, Take your time!

**Pull request merged**